src/components/Lesson.js

```diff
@@ -57,12 +55,22 @@ const Lesson = ({ route, navigation, dispatch, completedGrammars }) => {
     }
   };
 
+  const showFooter = () => {
+    if (completedGrammars[grammar.id]) {
```

Comment:
since you use this `completedGrammars[grammar.id]` in so many places I think it could be a variable.